SQL/New_patches/2018-01-17-normalisation_visit.sql

```diff
@@ -1,25 +1,27 @@
+ALTER TABLE project_rel DROP PRIMARY KEY, ADD COLUMN `ProjectSubprojectID` int(10) unsigned NOT NULL PRIMARY KEY AUTO_INCREMENT FIRST;
+ALTER TABLE project_rel ADD CONSTRAINT UK_project_rel_ProjectID_SubprojectID UNIQUE KEY (ProjectID, SubprojectID);
+
 CREATE TABLE `visit` (
   `VisitID` int(10) unsigned NOT NULL auto_increment,
   `VisitName` varchar(100) NOT NULL,
-  CONSTRAINT `visit_PK` PRIMARY KEY (`VisitID`),
-  CONSTRAINT `visit_name_UK` UNIQUE KEY (`VisitName`)
+  CONSTRAINT `PK_visit` PRIMARY KEY (`VisitID`),
+  CONSTRAINT `UK_visit_name` UNIQUE KEY (`VisitName`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
 
 CREATE TABLE `visit_project_subproject_rel` (
+  `VisitProjectSubprojectID` int(10) unsigned NOT NULL AUTO_INCREMENT,
   `VisitID` int(10) unsigned NOT NULL,
-  `ProjectID` int(2) NOT NULL,
-  `SubprojectID` int(10) unsigned NOT NULL,
-  CONSTRAINT visit_project_subproject_rel_PK PRIMARY KEY (`VisitID`, `ProjectID`, `SubprojectID`),
-  CONSTRAINT visit_project_subproject_rel_VisitID_visit_VisitID_FK FOREIGN KEY (`VisitID`) 
+  `ProjectSubprojectID` int(10) unsigned NOT NULL,
```

Comment:
I find the name of this column confusing. Can we call it something like a "ProjectSubprojectPair"?

Reply:
I dont like that
`ProjectSubprojectTuple` ??

Reply:
I'm fine with "Tuple" too.

Reply:
Made the change and added an amendment to the SQL modeling standards since that breaks the rules

https://github.com/aces/Loris/pull/4493/files#diff-190c4f51875cb53f0ebafdc6ae74b336R65